.github/workflows/build.yml

```diff
@@ -61,17 +76,138 @@ jobs:
               exit 1
               ;;
           esac
-          mkdir artifacts
-          wget -P "artifacts" "$downloadURL"
+          wget -P "temp" "$downloadURL"
+        working-directory: ravendb/scripts/linux/pkg/deb
+
+      - name: Build deb package
+        run: |
+          chmod a+x *.sh
+          case "${{ matrix.arch }}" in
+            "arm32v7")
+              arch="armhf"
+              ;;
+            "arm64v8")
+              arch="arm64"
+              ;;
+            "x64")
+              arch="amd64"
+              ;;
+            *)
+              echo "Unsupported architecture"
+              exit 1
+              ;;
+          esac
+          source ./set-ubuntu-${{ matrix.ubuntuName }}.sh
+          source ./set-raven-platform-${arch}.sh
+          ./build-deb.sh
+          echo "DEB_ARCH=$arch" >> $GITHUB_ENV
+          echo "UBUNTU_VERSION=$DISTRO_VERSION" >> $GITHUB_ENV
+        working-directory: ravendb/scripts/linux/pkg/deb
+        env:
+          RAVENDB_VERSION: ${{ inputs.version }}
+
+      - name: Upload deb package to artficats
+        uses: actions/upload-artifact@v3
+        with:
+          name: package-${{ matrix.ubuntuName }}-${{ matrix.arch }}
+          path: ravendb/scripts/linux/pkg/deb/dist/${{ env.UBUNTU_VERSION }}/ravendb*.deb
+
+      - name: Upload deb package to s3
+        run: |
+          set -e
+          set -x
+          #export DRY_RUN=1
+
+          PACKAGES_DIST_DIR=ravendb/scripts/linux/pkg/deb/dist
+
+          ls -R $PACKAGES_DIST_DIR
+
+          if [ "${{ env.IS_NIGHTLY }}" == "true" ]; then
+
+            NIGHTLY_VERSION_DATE_PART=$(date "+%Y%M%d-%H%m" -u)
+            export BUILD_VERSION="$NIGHTLY_VERSION_DATE_PART-Nightly"
+          elif [ "${{ env.IS_STABLE }}" == "true" ]; then
+            export BUILD_VERSION="${{ inputs.triggering_build_number }}"
+          else
+            export BUILD_VERSION="${{ inputs.triggering_build_number }}-${{ env.TRIGGERING_BUILD_TYPE }}"

```

Comment:
We talked that triggering_build_number is something different and you need something else.

Reply:
CHenged.